test/test_serialization.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import annotations
```

Comment:
Just FYI, there is already a `test/utils/test_serialization.py`. I don't mind having more tests, obviously.
This can stay here for now, and I can move things around / merge stuff later if need be.

Reply:
I've moved to the end of `test/utils/test_serialization.py`.